### PR TITLE
#29 fixes and test

### PR DESCRIPTION
**Pull request by Ekstazi**
Test and fix for #29

### CODE REVIEW

**Comment by Ekstazi** [comment]
I don't why phpunit on php 7.1 and 7.2 fails with segmentation failed. But i run each test individually and passed successfully

**Comment by Ekstazi** [comment]
Finally fixed segmentation failed in travis ci

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
+.idea
```
Comment (staabm) [inline .gitignore]: IDE related ignore rules should be part of your global git config within your development environment

File: .travis.yml [hunk 2]
```diff
@@ -25,7 +25,7 @@ install:
   - chmod +x coveralls.phar
 
 script:
-  - phpdbg -qrr vendor/bin/phpunit --coverage-text --coverage-clover build/logs/clover.xml
```
Comment (kelunik) [inline .travis.yml]: Disabling the GC is not really a solution. Maybe we should use xdebug again?

File: .gitignore (hunk 1 above)
Comment (Ekstazi) [reply .gitignore]: Thx. Fixed

File: .travis.yml (hunk 2 above)
Comment (Ekstazi) [reply .travis.yml]: @kelunik Can you explain please about xdebug ?

Comment (trowski) [reply .travis.yml]: Xdebug caused compatibility issues on travis in the past, but those may have been resolved. Remove the `before_install` section of `.travis.yml` and remove the portions of this line preceding `vendor/bin/phpunit …`.

File: lib/Internal/Posix/Runner.php [hunk 3]
```diff
@@ -156,6 +156,10 @@ public function kill(ProcessHandle $handle)
         if (!\proc_terminate($handle->proc, 9)) { // Forcefully kill the process using SIGKILL.
             throw new ProcessException("Terminating process failed");
         }
+        $handle->pidDeferred->promise()->onResolve(function ($error, $pid) {
+            // ignore errors because process not always detached
+            \posix_kill($pid, 9);
```
Comment (trowski) [inline lib/Internal/Posix/Runner.php]: The function should not call `posix_kill()` if `$pid` is null (i.e., there was an error starting the process).

File: .travis.yml (hunk 2 above)
Comment (Ekstazi) [reply .travis.yml]: I did some investigation. Segmentation fault doesn't depend on xdebug. The error occured when phpunit call testMultipleExecution. I'll try to fix it

Comment (Ekstazi) [reply .travis.yml]: I finally fixed it.

File: lib/Internal/Posix/Runner.php (hunk 3 above)
Comment (Ekstazi) [reply lib/Internal/Posix/Runner.php]: Fixed

**Pull request closed without merging**